### PR TITLE
Add a `normalize` function to Pine Type modules

### LINKED ISSUE #197: Pine types should have a 'normalize' function
Pine types should have a normalize function that accepts a value and tries to normalize to its intended type.
For example, the `Boolean` type should normalize a string value like `True`, `False`, `1`, and `0` into a true boolean and return it.

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #197

change-type: minor

### CODE REVIEW

File: src/components/PineTypes/Boolean.js [hunk 1]
```diff
@@ -20,6 +21,25 @@ export const rules = {
 
 export const validate = isBoolean
 
+export const normalize = value => {
+  if (isString(value)) {
+    switch (value.toLowerCase()) {
+    case 'true':
```
Comment (thgreasi) [inline src/components/PineTypes/Boolean.js]: So, does prettier like aligning the `case`s right below the `switch`!?!?!? :alien: :satellite:

File: src/components/PineTypes/CaseInsensitiveText.js [hunk 2]
```diff
@@ -28,6 +28,8 @@ export const rules = {
 
 export const validate = isString
 
+export const normalize = value => value.toString()
```
Comment (thgreasi) [inline src/components/PineTypes/CaseInsensitiveText.js]: Would it make sense to we make it safe to run against null/undefined values (here and below)?

Comment (LucianBuzzo) [reply src/components/PineTypes/CaseInsensitiveText.js]: yes indeed, I'll add tests for null and undefined to each of the types
